### PR TITLE
[Intel GPU] DeepSeek V4 9/N: use sgl-kernel implementation of hadamard_transform on XPU

### PR DESCRIPTION
**Pull request by polisettyvarma**
hadamard_transform is CUDA/HIP-only, so importing it from rotate_activation() fails on Intel XPU with ImportError when the NSA indexer path runs (e.g. DeepSeek-V4 compressed attention).

Add an iterative Walsh-Hadamard transform in pure torch and dispatch to it when _is_xpu is True. Same contract as the fused op: operates on the last dim, requires power-of-two size, multiplies by the supplied scale. CUDA/HIP/SM103/NPU paths are unchanged.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #29227863598](https://github.com/sgl-project/sglang/actions/runs/29227863598)
Latest PR Test (Extra): :x: [Run #29227863558](https://github.com/sgl-project/sglang/actions/runs/29227863558)

### CODE REVIEW

File: test/manual/test_hadamard.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import math
```
Comment (jianan-gu) [inline test/manual/test_hadamard.py]: The integration LGTM, while the other UTs here in the folder is more like for feature/serving level tests, can we move such UT in kernel side?

Comment (polisettyvarma) [reply test/manual/test_hadamard.py]: done

File: python/sglang/srt/layers/attention/dsa/dsa_indexer.py [hunk 2]
```diff
@@ -276,10 +278,32 @@ def topk_transform(
...
+    leading = x.shape[:-1]
+    out = x.reshape(-1, n).clone()
+    h = 1
+    while h < n:
+        out = out.view(-1, n // (2 * h), 2, h)
+        a = out[:, :, 0, :]
+        b = out[:, :, 1, :]
+        out = torch.stack((a + b, a - b), dim=2).view(-1, n)
+        h *= 2
+    return out.view(*leading, n) * scale
```
[the remark this answers is not in the document]
Comment (polisettyvarma) [reply python/sglang/srt/layers/attention/dsa/dsa_indexer.py]: tried and failed

**Review by jianan-gu — APPROVED** [review]

**Review by mingfeima — APPROVED** [review]
